str_analysis/call_non_ref_pathogenic_motifs.py

```diff
@@ -0,0 +1,978 @@
+#!/usr/bin/env python3
+
+DESCRIPTION = """This script takes a bam or cram file and determines which motifs are present at known pathogenic loci 
+(such as RFC1, BEAN1, DAB1, etc.) where several motifs are known to segregate in the population. It then optionally runs
+ ExpansionHunterDenovo and/or ExpansionHunter on the detected motifs and gathers relevant fields from their outputs. It
+ can also then run REViewer to generate read visualization images. 
+Finally it outputs a json file per locus with collected information as well as a "call" field indicating whether 
+pathogenic motifs were detected.
+"""
+
+import argparse
+import ast
+import collections
+import gzip
+import json
+import os
+import pkgutil
+import re
+import subprocess
+from pprint import pformat, pprint
+
+import pysam
+
+from str_analysis.utils.canonical_repeat_unit import \
+    compute_canonical_repeat_unit
+from str_analysis.utils.ehdn_info_for_locus import parse_ehdn_info_for_locus
+from str_analysis.utils.misc_utils import parse_interval
+from str_analysis.utils.most_frequent_repeat_unit import \
+    compute_most_frequent_repeat_unit
+
+LOCUS_INFO = json.loads(pkgutil.get_data(__name__, "data/locus_info.json"))
+OFFTARGET_REGIONS = json.loads(gzip.decompress(pkgutil.get_data(__name__, "data/offtarget_regions.json.gz")))
+
+MARGIN = 7   # base pairs
+FLANK_SIZE = 2000   # base pairs
+MIN_MAPQ = 3
+MIN_FRACTION_OF_BASES_COVERED = 0.7
+
+NORMALIZE_TO_COVERAGE = 40
+
+GENOME_VERSION_ALIASES = {
+    "GRCh37": "37", "hg19": "37", "hg37": "37", "37": "37",
+    "GRCh38": "38", "hg38": "38", "38": "38",
+}
+
+
+PATHOGENIC_MOTIF_CALL = "PATHOGENIC MOTIF"
+BENIGN_MOTIF_CALL = "BENIGN MOTIF"
+UNCERTAIN_SIG_CALL = "MOTIF OF UNCERTAIN SIGNIFICANCE"
+
+NO_CALL1 = "NO CALL (no motif has sufficient read support)"
+
+PATHOGENIC_PATHOGENIC_CALL = f"{PATHOGENIC_MOTIF_CALL} / {PATHOGENIC_MOTIF_CALL}"
+BENIGN_BENIGN_CALL = f"{BENIGN_MOTIF_CALL} / {BENIGN_MOTIF_CALL}"
+UNCERTAIN_SIG_UNCERTAIN_SIG_CALL = f"{UNCERTAIN_SIG_CALL} / {UNCERTAIN_SIG_CALL}"
+
+BENIGN_PATHOGENIC_CALL = f"{BENIGN_MOTIF_CALL} / {PATHOGENIC_MOTIF_CALL}"
+PATHOGENIC_UNCERTAIN_SIG_CALL = f"{PATHOGENIC_MOTIF_CALL} / {UNCERTAIN_SIG_CALL}"
+BENIGN_UNCERTAIN_SIG_CALL = f"{BENIGN_MOTIF_CALL} / {UNCERTAIN_SIG_CALL}"
+
+
+def parse_args():
```

Comment:
I currently don't find type annotations to be that useful, and prefer just having types documented in docstrings. What about you?

Reply:
For me, the overhead of adding type hints currently outweighs the benefits and makes args more difficult to read in some cases by adding more text to the function definition line that's often obvious (eg. I already know the file_path arg is a string). It would be one thing if people were running a tool to catch type errors based on the type hints, but for documentation, I prefer just adding types to the docstring. 

Overall, for stylistic suggestions where we have differing preferences, I'll leave the code as is since it's in a separate repo.